ledger/ledgertestutils/ledgertestutils.go

```diff
@@ -33,7 +35,7 @@ import (
 
 // TmpLedger crteates ledger on top of temporary database.
 // Returns *ledger.Ledger andh cleanup function.
-func TmpLedger(t testing.TB, lr core.LogicRunner, dir string) (*ledger.Ledger, func()) {
+func TmpLedger(t testing.TB, lr core.LogicRunner, dir string) (*ledger.Ledger, func(), consensus.NodeKeeper) {
```

Comment:
Let's make it
```
func TmpLedger(t testing.TB, dir string, c core.Components) (*ledger.Ledger, func()) 
```
If c is nil, make a default one. Otherwise use provided components.